platforms/documentation/docs/build.gradle.kts

```diff
@@ -600,120 +611,117 @@ tasks.named("quickTest") {
-            excludeTestsMatching 'org.gradle.docs.samples.*.convention-plugins*check.sample'
-            excludeTestsMatching 'org.gradle.docs.samples.*.convention-plugins*sanityCheck.sample'
-            excludeTestsMatching 'org.gradle.docs.samples.*.incubating-publishing-convention-plugins*publish.sample'
-            excludeTestsMatching 'org.gradle.docs.samples.*.publishing-convention-plugins*publish.sample'
-            excludeTestsMatching 'org.gradle.docs.samples.*.snippet-configuration-cache-test-kit*configurationCacheTestKit.sample'
-            excludeTestsMatching 'org.gradle.docs.samples.*.snippet-developing-plugins-testing-plugins*testPlugin.sample'
+        if (System.getProperty("bundleGroovy4", "false") == "true") {
+            excludeTestsMatching("org.gradle.docs.samples.*.convention-plugins*check.sample")
+            excludeTestsMatching("org.gradle.docs.samples.*.convention-plugins*sanityCheck.sample")
+            excludeTestsMatching("org.gradle.docs.samples.*.incubating-publishing-convention-plugins*publish.sample")
+            excludeTestsMatching("org.gradle.docs.samples.*.publishing-convention-plugins*publish.sample")
+            excludeTestsMatching("org.gradle.docs.samples.*.snippet-configuration-cache-test-kit*configurationCacheTestKit.sample")
+            excludeTestsMatching("org.gradle.docs.samples.*.snippet-developing-plugins-testing-plugins*testPlugin.sample")
         }
     }
 
     filter {
         // TODO(https://github.com/gradle/gradle/issues/22538)
-        excludeTestsMatching "org.gradle.docs.samples.*.snippet-groovy-cross-compilation_*_crossCompilation.sample"
+        excludeTestsMatching("org.gradle.docs.samples.*.snippet-groovy-cross-compilation_*_crossCompilation.sample")
     }
 
-    if (isConfigurationCacheEnabledForDocsTests(project)) {
+    if (project.configurationCacheEnabledForDocsTests) {
         systemProperty("org.gradle.integtest.samples.cleanConfigurationCacheOutput", "true")
         systemProperty("org.gradle.integtest.executer", "configCache")
 
         filter {
             // Configuration cache samples enable configuration cache explicitly. We're not going to run them with the configuration cache executer.
-            excludeTestsMatching "org.gradle.docs.samples.*.snippet-configuration-cache-*.sample"
-            excludeTestsMatching "*WithoutCC*.sample"
-
-            // Projects generated by gradle init enable configuration cache by default, no need to run them again.
-            excludeTestsMatching "org.gradle.docs.samples.*building-*.sample"
+            excludeTestsMatching("org.gradle.docs.samples.*.snippet-configuration-cache-*.sample")
+            excludeTestsMatching("*WithoutCC*.sample")
 
             // These tests cover features that are not planned to be supported in the first stable release of the configuration cache.
-            def testsForUnsupportedFeatures = [
+            val testsForUnsupportedFeatures = arrayOf(
```

Comment:
💅  It's more canonical to use `listOf` in cases like this in Kotlin. (there is a couple of these below too)